sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
Just did a quick check. Except `groupingSetToSQL`, the other nodes are not converting the `Project` operator to SQL. That is the reason why we have to add a special handling for `BroadcastHint`. Then, maybe we can add an extra dummy `Project` in `groupingSetToSQL`. If so, we can avoid adding the special handling for `BroadcastHint`